src/cogent3/core/alignment.py

```diff
@@ -2073,8 +2073,12 @@ def remapped_to(self, map):
         result = Aligned(map[self.map.inverse()].inverse(), self.data)
         return result
 
-    def get_annotations_matching(self, alignment, annotation_type="*", *args):
-        for annot in self.data.get_annotations_matching(annotation_type, *args):
+    def get_annotations_matching(
```

Comment:
We need to support the extended querying (e.g. `extend_query=True`) so add `**kwargs` to this method and the called method.